### PR TITLE
v1.12 Backports 2023-12-07

### PR DESCRIPTION
**Pull request by nbusseneau**
* [x] #29623 (@pchaigno)
 * [x] #29653 (@brb)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 29623 29653
```

### CODE REVIEW

**Review by brb — APPROVED** [review]
Thanks!

**Comment by nbusseneau** [comment]
@pchaigno Conflicts were hit while backporting your commits, please check backporter's notes for details :)

**Review by pchaigno — APPROVED** [review]

**Review by ti-mo — APPROVED** [review]

**Comment by giorio94** [comment]
> https://github.com/cilium/cilium/actions/runs/7130488093/job/19416964192 => `unknown flag: --conn-disrupt-dispatch-interval` 🤔
> 
> I suspect the conn-disrupt changes actually depend on something else that's missing. It's probably the CLI version.

Bumped the Cilium CLI to the latest version (v0.15.17), which should fix the issue).

**Comment by giorio94** [comment]
Rebased onto the tip of v1.12 to pick https://github.com/cilium/cilium/commit/0186818f8cfa4b4e0043a838cb01096988690033 and fix Travis.

**Comment by giorio94** [comment]
/test-backport-1.12

**Comment by giorio94** [comment]
IPsec failures are https://github.com/cilium/cilium/issues/29805

**Comment by giorio94** [comment]
/test-1.20-4.9

**Comment by giorio94** [comment]
Marking as blocked while IPSec the new failures in Cilium IPSec upgrade get investigated: https://github.com/cilium/cilium/pull/29702#issuecomment-1854585443

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -98,6 +98,15 @@ jobs:
...
             ipv6: 'false' # until https://github.com/cilium/cilium/issues/26944 resolved
 
+          - config: '6.1'
+            # renovate: datasource=docker depName=quay.io/lvh-images/kind
+            kernel: '6.1-20231026.065108'
+            kube-proxy: 'iptables'
+            kpr: 'disabled'
+            tunnel: 'vxlan'
+            encryption: 'ipsec'
+            endpoint-routes: 'false'
```
Comment (brb) [inline .github/workflows/tests-ipsec-upgrade.yaml:108]: @nbusseneau Could you append the following to the commit which added this config?
```
# https://github.com/cilium/cilium/issues/29805#issuecomment-1855600171
ipv6: 'false'
```

**Comment by brb** [comment]
/test-backport-1.12

**Pull request merged**